mithril-signer/src/metrics/service.rs

```diff
@@ -1,278 +1,147 @@
-    SIGNER_REGISTRATION_TOTAL_SINCE_STARTUP_METRIC_NAME,
-};
-
-/// Type alias for a metric name.
-pub type MetricName = str;
-
-/// Type alias for a counter value.
-type CounterValue = u32;
-
-/// Metrics service which is responsible for recording and exposing metrics.
-pub struct MetricsService {
-    registry: Registry,
-    logger: Logger,
-    signer_registration_success_since_startup_counter: Box<Counter>,
-    signer_registration_total_since_startup_counter: Box<Counter>,
-    signer_registration_success_last_epoch_gauge: Box<Gauge>,
-    signature_registration_success_since_startup_counter: Box<Counter>,
-    signature_registration_total_since_startup_counter: Box<Counter>,
-    signature_registration_success_last_epoch_gauge: Box<Gauge>,
-    runtime_cycle_success_since_startup_counter: Box<Counter>,
-    runtime_cycle_total_since_startup_counter: Box<Counter>,
-}
-
+use mithril_metric::metric::{CounterValue, MetricCollector, MetricCounter, MetricGauge};
+
+build_metrics_service!(
+    MetricsService,
+    signer_registration_success_since_startup_counter:MetricCounter(
+        "mithril_signer_signer_registration_success_since_startup",
+        "Number of successful signer registrations since startup on a Mithril signer node"
+    ),
+    signer_registration_total_since_startup_counter:MetricCounter(
+        "mithril_signer_signer_registration_total_since_startup",
+        "Number of signer registrations since startup on a Mithril signer node"
+    ),
+    signer_registration_success_last_epoch_gauge:MetricGauge(
+        "mithril_signer_signer_registration_success_last_epoch",
+        "Latest epoch at which signer successfully registered on a Mithril signer node"
+    ),
+    signature_registration_success_since_startup_counter:MetricCounter(
+        "mithril_signer_signature_registration_success_since_startup",
+        "Number of successful signature registrations since startup on a Mithril signer node"
+    ),
+    signature_registration_total_since_startup_counter:MetricCounter(
+        "mithril_signer_signature_registration_total_since_startup",
+        "Number of signature registrations since startup on a Mithril signer node"
+    ),
+    signature_registration_success_last_epoch_gauge:MetricGauge(
+        "mithril_signer_signature_registration_success_last_epoch",
+        "Latest epoch at which signature successfully registered on a Mithril signer node"
+    ),
+    // Runtime cycle metrics
+    runtime_cycle_success_since_startup_counter:MetricCounter(
+        "mithril_signer_runtime_cycle_success_since_startup",
+        "Number of successful runtime cycles since startup on a Mithril signer node"
+    ),
+    runtime_cycle_total_since_startup_counter:MetricCounter(
+        "mithril_signer_runtime_cycle_total_since_startup",
+        "Number of runtime cycles since startup on a Mithril signer node"
+    )
+
+);
+
+// TODO these functions are kept while changes are made for all uses
```

Comment:
Is this TODO still relevant ?

Reply:
I removed all obsolete function and associated tests